ddtrace/context.py

```diff
@@ -34,7 +34,7 @@ class Context(object):
     """
 
     _partial_flush_enabled = asbool(get_env("tracer", "partial_flush_enabled", default=False))
-    _partial_flush_min_spans = int(get_env("tracer", "partial_flush_min_spans", default=500))
+    _partial_flush_min_spans = int(get_env("tracer", "partial_flush_min_spans", default=500))  # type: ignore[arg-type]
```

Comment:
It has to do with the fact that `get_env` pulls `default` from `kwargs` so type inference gets mighty confused:

~~~
error: Argument 1 to "int" has incompatible type "Union[str, SupportsInt, _SupportsIndex, _SupportsTrunc, None]"; expected "Union[str, SupportsInt, _SupportsIndex, _SupportsTrunc]"
~~~
So here `int` is never getting a `None` because `default` is passed in to `get_env`

Reply:
Ok.
It might be better to pass the type to `get_env` rather than ignoring the typing info here.

Reply:
That's fine by me. @brettlangdon any thoughts?

Reply:
I wonder if we have `get_env` always return `Optional[str]` and `default` must be `Optional[str]`, we'll be wasting some time converting a default of `false` with `asbool` but it'll give us a consistent experience with type usage on `get_env`

... or we use a generic and have the response be `Optional[T]` where `default` is also `Optional[T]` or... something, /shrug

Reply:
It looks like whatever I try I end up with having to make changes that perhaps are not worth these few type hints